docs/1-introduction/1-1-getting-started/1-1-3-hello-triangle.md

```diff
@@ -1,11 +1,456 @@
+## Compiling shaders
+
+Now that we wrote our shader code and saved it somewhere, we have to feed this to the GPU, to do that we'll have our D3DCompiler get to work.
+
+First, we will declare some functions that will help us compile our shaders more quickly.
+
+### HelloTriangle.hpp
+```cpp
+bool CompileShader(
+    const std::wstring_view fileName,
+    const std::string_view entryPoint,
+    const std::string_view profile,
+    ComPtr<ID3D10Blob>& shaderBlob) const;
+
+[[nodiscard]] ComPtr<ID3D11VertexShader> CreateVertexShader(
+    const std::wstring_view fileName,
+    ComPtr<ID3D10Blob>& vertexShaderBlob) const;
+
+[[nodiscard]] ComPtr<ID3D11PixelShader> CreatePixelShader(std::wstring_view fileName) const;
+```
+In order we have:
+
+`CompileShader`: This function is the core for compiling shaders, it requires 3 input parameters:
+
+- `fileName`: is the path of the shader file we want to compile.
+- `entryPoint`: is the name of the function where the shader begins execution.
+- `profile`: which is basically the version of HLSL we want to use, the higher the profile number, the more features there are.
+
+And one output parameter:
+
+- `shaderBlob`: the blob were our compiled code will be stored. A blob is just a fancy buffer which D3D11 can use for specific purposes.
+
+Then:
+`CreateVertexShader`: This function helps us create specifically a `ID3D11VertexShader`, it only requires the shader path and a `ID3D10Blob`. We need to pass a blob ourselves because we'll need the VS's blob later.
+
+`CreatePixelShader`: It does the same thing that `CreateVertexShader` does, except we don't need to pass a `ID3D10Blob` here.
+
+Now that we know how our new members look, we will see how we implemented them.
+### HelloTriangle.cpp
+First things first, let's see `CompileShader`:
+#### CompileShader
+```cpp
+bool HelloTriangleApplication::CompileShader(
+    const std::wstring_view fileName,
+    const std::string_view entryPoint,
+    const std::string_view profile,
+    ComPtr<ID3D10Blob>& shaderBlob) const
+{
+    constexpr UINT compileFlags = D3DCOMPILE_ENABLE_STRICTNESS;
+
+    ComPtr<ID3D10Blob> tempShaderBlob = nullptr;
+    ComPtr<ID3D10Blob> errorBlob = nullptr;
+    if (FAILED(D3DCompileFromFile(
+        fileName.data(),
+        nullptr,
+        D3D_COMPILE_STANDARD_FILE_INCLUDE,
+        entryPoint.data(),
+        profile.data(),
+        compileFlags,
+        0,
+        &tempShaderBlob,
+        &errorBlob)))
+    {
+        std::cout << "D3D11: Failed to read shader from file\n";
+        if (errorBlob != nullptr)
+        {
+            std::cout << "D3D11: With message: " << static_cast<const char*>(errorBlob->GetBufferPointer()) << "\n";
+        }
+
+        return false;
+    }
+
+    shaderBlob = std::move(tempShaderBlob);
+    return true;
+}
+```
+
+We start by creating two `ID3D10Blob`s, we will need a temporary blob, where we'll load our shader file and an error blob, which will contain our error messages, if any.
```

Comment:
Maybe also turn it into a code block hyperlink

[`ID3D10Blob`](msdn-link-to-id3d10blob)

Maybe also add a sentence somewhere, explaining why blob is ID3D10Blob and not ID3D11Blob